CLI/controllers/controller.go

```diff
@@ -0,0 +1,11 @@
+package controllers
+
+var C = Controller{
+	API:     API,
+	Ogree3D: Ogree3D,
+}
+
+type Controller struct {
+	API     APIPort
+	Ogree3D Ogree3DPort
+}
```

Comment:
Im not a big fan of these extra imbricated global variables, it is hard to follow. I should have made the remark for the unity connection PR (because of the Ogree3DConnection one), but i did not pay much attention to it. 
We already have one big global variable for the state of the CLI. I dont see why the "isConnected" boolean of Ogree3DConnection would not be part of that state. Here APIPort does not have any member so I do not see the point either. We can make new packages if we want to the functions to be better organized.

Reply:
All these changes are to ensure the testability of the cli. To be able to make tests that do not depend on the api, mocks are necessary, to be able to simulate the response that the api would give for certain request. You can see examples of this in the tests I am adding in delete_test.go and update_test.go. In future pull requests I am adding many more tests. It is necessary to move from function schema to structures because a function call is impossible to mock. Instead, the functions are passed to structures with methods and an interface is created with all the methods of that structure, to be able to create the mocks (see interface APIPort and mocks/controllers/APIPort.go). The mocks are generated automatically with mockery.
Anyway, the global variables should not be necessary, since the objects could create their dependencies (and use dependency injection in case of complexity), but the problem is that to achieve this you have to pass all the functions to the new schema and it is a change that will have to be done progressively.
On the other hand, you could avoid global variables using a singleton pattern, but I think that the complexity of the application does not merit doing that work (there are only 3 global variables).

Reply:
Ok I understand better now. Maybe it will be worth spending time trying to uniformize the pattern across the whole project, or find another way to mock functions. But I agree that it can be done progressively.